### ISSUE TITLE
PHP Fatal error:  Uncaught ValueError: DOMDocument::loadXML(): Argument #1 ($source) must not be empty in

### ISSUE THREAD

**Issue by janevsky** [pkp/pkp-lib#9535]
**Describe the bug**
We published another issue yesterday night and a new submission, this morning i wanted to reset the counter on the scheduled tasks and did the following:
1. Reload Scheduled Tasks from acron plugin
2. Deleted 8 rows in scheduled_tasks from the db
3. Hard refresh(shift+f5) on my journal home page so the scheduled tasks load again
4. After the first refresh got an invalid XML error on the bottom on my page (didn't get a screenshot)

**To Reproduce**
I believe that it will be the same as i described above, did not try, i will leave it like this until tomorrow to see if it happens again.

**What application are you using?**
 OJS version 3.3.0-14 
PHP version 8.0.30

**Additional information**
Refreshing the page once got me the invalid XML error on the bottom, i've checked the scheduled_tasks it contained only 2 out of 8 rows, plugins.generic.usageStats.UsageStatsLoader and plugins.generic.pln.classes.tasks.Depositor, the rest were loaded on my second refresh.
![image](https://github.com/pkp/pkp-lib/assets/63719545/e192278d-48f7-4762-9823-1571cd5d4a50)
The submission successfully deposited the doi and the doi link is working. The view/download counter changed also.

Not sure if the php warnings are related to this but i will post what i got when i reloaded.
PHP error log:
`[24-Nov-2023 06:49:57 Europe/Ljubljana] PHP Warning:  Undefined array key 0 in /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php on line 85
[24-Nov-2023 06:49:57 Europe/Ljubljana] PHP Warning:  Undefined array key 0 in /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php on line 85
[24-Nov-2023 06:49:57 Europe/Ljubljana] PHP Warning:  Attempt to read property "documentElement" on null in /home/XXX/domains/XXX/public_html/plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php on line 204
[24-Nov-2023 06:49:57 Europe/Ljubljana] PHP Warning:  Cannot modify header information - headers already sent by (output started at /home/XXX/domains/XXX/public_html/lib/pkp/plugins/generic/acron/PKPAcronPlugin.inc.php:219) in /home/XXX/domains/XXX/public_html/lib/pkp/classes/plugins/ImportExportPlugin.inc.php on line 171
[24-Nov-2023 06:49:57 Europe/Ljubljana] PHP Warning:  Undefined variable $exportXml in /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php on line 328
[24-Nov-2023 06:49:57 Europe/Ljubljana] PHP Fatal error:  Uncaught ValueError: DOMDocument::loadXML(): Argument #1 ($source) must not be empty in /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php:411
Stack trace:
#0 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php(411): DOMDocument->loadXML()
#1 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php(328): DepositPackage->_cleanFileList()
#2 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php(540): DepositPackage->generatePackage()
#3 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/tasks/Depositor.inc.php(225): DepositPackage->packageDeposit()
#4 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/tasks/Depositor.inc.php(127): Depositor->_processNeedPackaging()
#5 /home/XXX/domains/XXX/public_html/lib/pkp/classes/scheduledTask/ScheduledTask.inc.php(146): Depositor->executeActions()
#6 /home/XXX/domains/XXX/public_html/lib/pkp/plugins/generic/acron/PKPAcronPlugin.inc.php(259): ScheduledTask->execute()
#7 [internal function]: PKPAcronPlugin->shutdownFunction()
#8 {main}
  thrown in /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php on line 411`

If more information is needed i can provide.

**Comment by asmecher** [comment]
@jonasraoni, the error message seems to be coming from the PKP|PN plugin. Can you have a look?

**Comment by jonasraoni** [comment]
@asmecher The problems on the PN plugin have been fixed, we just need to merge the pending PRs and do a release.

**Comment by jonasraoni** [comment]
@janevsky Thanks for the report! Could you show us what do you have on the line 85 of this file `/home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php`? Looks like it's not the same that I see here: https://github.com/pkp/pkp-lib/blob/668ad21beea676ef3eda92b03a5ac3df4b062d78/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php#L85C1-L85C1

I guess you have this code, right?
`$submissionFileNode->setAttribute('stage', $stageToName[$submissionFile->getFileStage()]);`

If yes, then looks like your submission file has with an invalid stage, and I'm not sure it's possible to fix it properly 🤔

@asmecher What do you think about updating empty/invalid file stages to `SUBMISSION_FILE_SUBMISSION` (or another not very important constant)? Or if this is known to be the result of a half-baked operation, then we can remove such occurrences on the upgrade.

**Comment by janevsky** [comment]
> @janevsky Thanks for the report! Could you show us what do you have on the line 85 of this file `/home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php`? Looks like it's not the same that I see here: https://github.com/pkp/pkp-lib/blob/668ad21beea676ef3eda92b03a5ac3df4b062d78/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php#L85C1-L85C1
> 
> I guess you have this code, right? `$submissionFileNode->setAttribute('stage', $stageToName[$submissionFile->getFileStage()]);`
> 
> If yes, then looks like your submission file has with an invalid stage, and I'm not sure it's possible to fix it properly 🤔
> 
> @asmecher What do you think about updating empty/invalid file stages to `SUBMISSION_FILE_SUBMISSION` (or another not very important constant)? Or if this is known to be the result of a half-baked operation, then we can remove such occurrences on the upgrade.
> 
> And as a fix for the Native plugin, we can skip such files.

@jonasraoni That's exacaly what i have on line 85, here is the full code inside SubmissionFileNativeXmlFilter.inc.php:
`<?php

/**
 * @file plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php
 *
 * Copyright (c) 2014-2021 Simon Fraser University
 * Copyright (c) 2000-2021 John Willinsky
 * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
 *
 * @class SubmissionFileNativeXmlFilter
 * @ingroup plugins_importexport_native
 *
 * @brief Base class that converts a submissionFile to a Native XML document
 */

import('lib.pkp.plugins.importexport.native.filter.NativeExportFilter');

class SubmissionFileNativeXmlFilter extends NativeExportFilter {
	/**
	 * Constructor
	 * @param $filterGroup FilterGroup
	 */
	function __construct($filterGroup) {
		$this->setDisplayName('Native XML submission file export');
		parent::__construct($filterGroup);
	}


	//
	// Implement template methods from PersistableFilter
	//
	/**
	 * @copydoc PersistableFilter::getClassName()
	 */
	function getClassName() {
		return 'lib.pkp.plugins.importexport.native.filter.SubmissionFileNativeXmlFilter';
	}

	//
	// Implement template methods from Filter
	//
	/**
	 * @see Filter::process()
	 * @param $submissionFile SubmissionFile
	 * @return DOMDocument
	 */
	function &process(&$submissionFile) {
		// Create the XML document
		$doc = new DOMDocument('1.0');
		$doc->preserveWhiteSpace = false;
		$doc->formatOutput = true;
		$deployment = $this->getDeployment();
		$rootNode = $this->createSubmissionFileNode($doc, $submissionFile);
		$doc->appendChild($rootNode);
		$rootNode->setAttributeNS('http://www.w3.org/2000/xmlns/', 'xmlns:xsi', 'http://www.w3.org/2001/XMLSchema-instance');
		$rootNode->setAttribute('xsi:schemaLocation', $deployment->getNamespace() . ' ' . $deployment->getSchemaFilename());

		return $doc;
	}

	//
	// SubmissionFile conversion functions
	//
	/**
	 * Create and return a submissionFile node.
	 * @param $doc DOMDocument
	 * @param $submissionFile SubmissionFile
	 * @return DOMElement
	 */
	function createSubmissionFileNode($doc, $submissionFile) {
		$deployment = $this->getDeployment();
		$context = $deployment->getContext();
		$stageToName = array_flip($deployment->getStageNameStageIdMapping());
		$genreDao = DAORegistry::getDAO('GenreDAO'); /* @var $genreDao GenreDAO */
		$genre = $genreDao->getById($submissionFile->getData('genreId'));
		$userDao = DAORegistry::getDAO('UserDAO'); /* @var $userDao UserDAO */
		$uploaderUser = $userDao->getById($submissionFile->getData('uploaderUserId'));

		// Create the submission_file node and set metadata
		$submissionFileNode = $doc->createElementNS($deployment->getNamespace(), $this->getSubmissionFileElementName());
		$submissionFileNode->setAttribute('id', $submissionFile->getId());
		$submissionFileNode->setAttribute('created_at', strftime('%Y-%m-%d', strtotime($submissionFile->getData('createdAt'))));
		$submissionFileNode->setAttribute('date_created', $submissionFile->getData('dateCreated'));
		$submissionFileNode->setAttribute('file_id', $submissionFile->getData('fileId'));
		$submissionFileNode->setAttribute('stage', $stageToName[$submissionFile->getFileStage()]);
		$submissionFileNode->setAttribute('updated_at', strftime('%Y-%m-%d', strtotime($submissionFile->getData('updatedAt'))));
		$submissionFileNode->setAttribute('viewable', $submissionFile->getViewable()?'true':'false');
		if ($caption = $submissionFile->getData('caption')) {
			$submissionFileNode->setAttribute('caption', $caption);
		}
		if ($copyrightOwner = $submissionFile->getData('copyrightOwner')) {
			$submissionFileNode->setAttribute('copyright_owner', $copyrightOwner);
		}
		if ($credit = $submissionFile->getData('credit')) {
			$submissionFileNode->setAttribute('credit', $credit);
		}
		if ($directSalesPrice = $submissionFile->getData('directSalesPrice')) {
			$submissionFileNode->setAttribute('direct_sales_price', $directSalesPrice);
		}
		if ($genre) {
			$submissionFileNode->setAttribute('genre', $genre->getName($context->getPrimaryLocale()));
		}
		if ($language = $submissionFile->getData('language')) {
			$submissionFileNode->setAttribute('language', $language);
		}
		if ($salesType = $submissionFile->getData('salesType')) {
			$submissionFileNode->setAttribute('sales_type', $salesType);
		}
		if ($sourceSubmissionFileId = $submissionFile->getData('sourceSubmissionFileId')) {
			$submissionFileNode->setAttribute('source_submission_file_id', $sourceSubmissionFileId);
		}
		if ($terms = $submissionFile->getData('terms')) {
			$submissionFileNode->setAttribute('terms', $terms);
		}
		if ($uploaderUser) {
			$submissionFileNode->setAttribute('uploader', $uploaderUser->getUsername());
		}
		$this->createLocalizedNodes($doc, $submissionFileNode, 'creator', $submissionFile->getData('creator'));
		$this->createLocalizedNodes($doc, $submissionFileNode, 'description', $submissionFile->getData('description'));
		$this->createLocalizedNodes($doc, $submissionFileNode, 'name', $submissionFile->getData('name'));
		$this->createLocalizedNodes($doc, $submissionFileNode, 'publisher', $submissionFile->getData('publisher'));
		$this->createLocalizedNodes($doc, $submissionFileNode, 'source', $submissionFile->getData('source'));
		$this->createLocalizedNodes($doc, $submissionFileNode, 'sponsor', $submissionFile->getData('sponsor'));
		$this->createLocalizedNodes($doc, $submissionFileNode, 'subject', $submissionFile->getData('subject'));

		// If it is a dependent file, add submission_file_ref element
		if ($submissionFile->getData('fileStage') == SUBMISSION_FILE_DEPENDENT && $submissionFile->getData('assocType') == ASSOC_TYPE_SUBMISSION_FILE) {
			$fileRefNode = $doc->createElementNS($deployment->getNamespace(), 'submission_file_ref');
			$fileRefNode->setAttribute('id', $submissionFile->getData('assocId'));
			$submissionFileNode->appendChild($fileRefNode);
		}

		// Add pub-id plugins
		$this->addIdentifiers($doc, $submissionFileNode, $submissionFile);

		// Create the revision nodes
		$revisions = DAORegistry::getDAO('SubmissionFileDAO')->getRevisions($submissionFile->getId());
		foreach ($revisions as $revision) {
			$localPath = rtrim(Config::getVar('files', 'files_dir'), '/') . '/' . $revision->path;
			$revisionNode = $doc->createElementNS($deployment->getNamespace(), 'file');
			$revisionNode->setAttribute('id', $revision->fileId);
			$revisionNode->setAttribute('filesize', filesize($localPath));
			$revisionNode->setAttribute('extension', pathinfo($revision->path, PATHINFO_EXTENSION));

			if (array_key_exists('no-embed', $this->opts)) {
				$hrefNode = $doc->createElementNS($deployment->getNamespace(), 'href');
				if (array_key_exists('use-file-urls', $this->opts)) {
					$stageId = Services::get('submissionFile')->getWorkflowStageId($submissionFile);
					$dispatcher = Application::get()->getDispatcher();
					$request = Application::get()->getRequest();
					$params = [
						"submissionFileId" => $submissionFile->getId(),
						"submissionId" => $submissionFile->getData('submissionId'),
						"stageId" => $stageId,
					];
					$url = $dispatcher->url($request, ROUTE_COMPONENT, $context->getPath(), "api.file.FileApiHandler", "downloadFile", null, $params);
					$hrefNode->setAttribute('src', $url);
				} else {
					$hrefNode->setAttribute('src', $revision->path);
				}
				$hrefNode->setAttribute('mime_type', $revision->mimetype);
				$revisionNode->appendChild($hrefNode);
			} else {
				$embedNode = $doc->createElementNS($deployment->getNamespace(), 'embed', base64_encode(file_get_contents($localPath)));
				$embedNode->setAttribute('encoding', 'base64');
				$revisionNode->appendChild($embedNode);
			}

			$submissionFileNode->appendChild($revisionNode);
		}


		return $submissionFileNode;
	}

	/**
	 * Create and add identifier nodes to a submission node.
	 * @param $doc DOMDocument
	 * @param $revisionNode DOMElement
	 * @param $submissionFile SubmissionFile
	 */
	function addIdentifiers($doc, $revisionNode, $submissionFile) {
		$deployment = $this->getDeployment();

		// Ommiting the internal ID here because it is in the submission_file attribute

		// Add public ID
		if ($pubId = $submissionFile->getStoredPubId('publisher-id')) {
			$revisionNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'id', htmlspecialchars($pubId, ENT_COMPAT, 'UTF-8')));
			$node->setAttribute('type', 'public');
			$node->setAttribute('advice', 'update');
		}

		// Add pub IDs by plugin
		$pubIdPlugins = PluginRegistry::loadCategory('pubIds', true, $deployment->getContext()->getId());
		foreach ($pubIdPlugins as $pubIdPlugin) {
			$this->addPubIdentifier($doc, $revisionNode, $submissionFile, $pubIdPlugin);
		}
	}

	/**
	 * Add a single pub ID element for a given plugin to the document.
	 * @param $doc DOMDocument
	 * @param $revisionNode DOMElement
	 * @param $submissionFile SubmissionFile
	 * @param $pubIdPlugin PubIdPlugin
	 * @return DOMElement|null
	 */
	function addPubIdentifier($doc, $revisionNode, $submissionFile, $pubIdPlugin) {
		$pubId = $submissionFile->getStoredPubId($pubIdPlugin->getPubIdType());
		if ($pubId) {
			$deployment = $this->getDeployment();
			$revisionNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'id', htmlspecialchars($pubId, ENT_COMPAT, 'UTF-8')));
			$node->setAttribute('type', $pubIdPlugin->getPubIdType());
			$node->setAttribute('advice', 'update');
			return $node;
		}
		return null;
	}

	/**
	 * Get the submission file element name
	 */
	function getSubmissionFileElementName() {
		return 'submission_file';
	}
}


`
I did a compare, the full code seems the same.
AS you mentioned, the file is in an invalid stage? What do you mean? The submission is published and the doi deposited with success. 
Here is my PKP PN plugin status, tried with reset button on both when i've posted the bug here.
![image](https://github.com/pkp/pkp-lib/assets/63719545/732ddb1d-922d-46a4-a84d-b3f124990bd5)

This is the other day in the morning with the scheduled tasks:
`[25-Nov-2023 07:06:13 Europe/Ljubljana] PHP Warning:  Undefined array key 0 in /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php on line 85
[25-Nov-2023 07:06:13 Europe/Ljubljana] PHP Warning:  Undefined array key 0 in /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php on line 85
[25-Nov-2023 07:06:13 Europe/Ljubljana] PHP Warning:  Undefined array key 0 in /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php on line 85
[25-Nov-2023 07:06:13 Europe/Ljubljana] PHP Warning:  Undefined array key 0 in /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php on line 85
[25-Nov-2023 07:06:13 Europe/Ljubljana] PHP Warning:  Attempt to read property "documentElement" on null in /home/XXX/domains/XXX/public_html/plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php on line 204
[25-Nov-2023 07:06:13 Europe/Ljubljana] PHP Warning:  Cannot modify header information - headers already sent by (output started at /home/XXX/domains/XXX/public_html/lib/pkp/plugins/generic/acron/PKPAcronPlugin.inc.php:219) in /home/XXX/domains/XXX/public_html/lib/pkp/classes/plugins/ImportExportPlugin.inc.php on line 171
[25-Nov-2023 07:06:14 Europe/Ljubljana] PHP Warning:  Undefined variable $exportXml in /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php on line 328
[25-Nov-2023 07:06:14 Europe/Ljubljana] PHP Fatal error:  Uncaught ValueError: DOMDocument::loadXML(): Argument #1 ($source) must not be empty in /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php:411
Stack trace:
#0 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php(411): DOMDocument->loadXML()
#1 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php(328): DepositPackage->_cleanFileList()
#2 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php(540): DepositPackage->generatePackage()
#3 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/tasks/Depositor.inc.php(225): DepositPackage->packageDeposit()
#4 /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/tasks/Depositor.inc.php(127): Depositor->_processNeedPackaging()
#5 /home/XXX/domains/XXX/public_html/lib/pkp/classes/scheduledTask/ScheduledTask.inc.php(146): Depositor->executeActions()
#6 /home/XXX/domains/XXX/public_html/lib/pkp/plugins/generic/acron/PKPAcronPlugin.inc.php(259): ScheduledTask->execute()
#7 [internal function]: PKPAcronPlugin->shutdownFunction()
#8 {main}
  thrown in /home/XXX/domains/XXX/public_html/plugins/generic/pln/classes/DepositPackage.inc.php on line 411`
If its a problem from my side please give me some guidance on what should i aim for to fix this.

**Comment by jonasraoni** [comment]
Thanks @janevsky! Yes, it's a problem that you'll have to fix there (or wait for the next OJS release) :)

I can proceed with a "fast fix", but before doing anything, I'll wait for feedback from @asmecher.

**Comment by janevsky** [comment]
@jonasraoni okay thank you, i will be waiting for a replay.

**Comment by janevsky** [comment]
update: There were no errors on the new submission, everything was done the same like the submission before.

**Comment by jonasraoni** [comment]
@janevsky I've left a workaround here, which will just skip files that have an invalid file stage (the "file stage" defines if the file is part of the copyediting, production, ...): https://github.com/pkp/pkp-lib/pull/9537/files

This query should bring the bad submission files:
```sql
select sf.submission_id, f.path
from submission_files sf
inner join files f on f.file_id = sf.file_id
where sf.file_stage = 0
```

You can take a look at the submission and the physical file... Then try to understand if the file is duplicated on the user interface (in case it was the result of a half-baked operation and the user re-uploaded) or if it's just went missing from one of the stages.

**Comment by janevsky** [comment]
@jonasraoni thank you for the workaround, I've updated the two files, i will keep an eye on the PHP error log when we publish a new submission. I believe it wont have any problems.

For the query, here are the results:
![image](https://github.com/pkp/pkp-lib/assets/63719545/d9ae1c8c-efcd-48f1-8c2b-ad7012883e2c)
I will use this query to check of the future submissions if we get wrong file stage id's and try to find the root of the problem.

Thank you very much!

**Comment by janevsky** [comment]
@jonasraoni now a new error poped, here is the php error log from this morning:
`[28-Nov-2023 07:30:39 Europe/Ljubljana] PHP Fatal error:  Uncaught TypeError: DOMNode::appendChild(): Argument #1 ($node) must be of type DOMNode, null given in /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php:54
Stack trace:
#0 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php(54): DOMNode->appendChild()
#1 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/classes/filter/Filter.inc.php(449): SubmissionFileNativeXmlFilter->process()
#2 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionNativeXmlFilter.inc.php(149): Filter->execute()
#3 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionNativeXmlFilter.inc.php(101): SubmissionNativeXmlFilter->addFiles()
#4 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/importexport/native/filter/ArticleNativeXmlFilter.inc.php(49): SubmissionNativeXmlFilter->createSubmissionNode()
#5 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionNativeXmlFilter.inc.php(65): ArticleNativeXmlFilter->createSubmissionNode()
#6 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/classes/filter/Filter.inc.php(449): SubmissionNativeXmlFilter->process()
#7 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php(203): Filter->execute()
#8 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php(108): IssueNativeXmlFilter->addArticles()
#9 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php(57): IssueNativeXmlFilter->createIssueNode()
#10 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/classes/filter/Filter.inc.php(449): IssueNativeXmlFilter->process()
#11 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/importexport/native/NativeImportExportPlugin.inc.php(289): Filter->execute()
#12 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/generic/pln/classes/DepositPackage.inc.php(314): NativeImportExportPlugin->exportIssues()
#13 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/generic/pln/classes/DepositPackage.inc.php(540): DepositPackage->generatePackage()
#14 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/generic/pln/classes/tasks/Depositor.inc.php(225): DepositPackage->packageDeposit()
#15 /home/jhrs/domains/jhrs.almamater.si/public_html/plugins/generic/pln/classes/tasks/Depositor.inc.php(127): Depositor->_processNeedPackaging()
#16 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/classes/scheduledTask/ScheduledTask.inc.php(146): Depositor->executeActions()
#17 /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/plugins/generic/acron/PKPAcronPlugin.inc.php(259): ScheduledTask->execute()
#18 [internal function]: PKPAcronPlugin->shutdownFunction()
#19 {main}
  thrown in /home/jhrs/domains/jhrs.almamater.si/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php on line 54`

**Comment by asmecher** [comment]
@jonasraoni,

> What do you think about updating empty/invalid file stages to `SUBMISSION_FILE_SUBMISSION` (or another not very important constant)? Or if this is known to be the result of a half-baked operation, then we can remove such occurrences on the upgrade.

I've looked through a number of databases from various versions of OJS and only found two databases with `file_stage_id = 0`. In trying to understand the source, I looked at 
- the OJS version used at the time (2.3.5-0 and 2.1.0-1 to 2.2.1-0)
- the type of role that uploaded the files (managers in one installation, and reviewers in the other)

I think tracking it down more specifically would require a deep-dive into OJS 2.x history, and I don't think that would be worthwhile. So while I can't say 100% that it won't cause side-effects, I'd support what you propose, Jonas!

**Comment by janevsky** [comment]
@jonasraoni Forgot to mention that I did a reset on the PKP PN plugin when I did the changes that you proposed, and the next morning i got the php error above and here is the status now:
![image](https://github.com/pkp/pkp-lib/assets/63719545/f6e6ee7b-85d2-40aa-9745-f8b858108d5a)
no.4 passed, but no.3 still has errors.

**Comment by jonasraoni** [comment]
@janevsky Sorry for the delay, but I guess you have a different issue now.

You can try to replace the following line in the file `lib/pkp/classes/xslt/XMLTypeDescription.inc.php`:
https://github.com/pkp/pkp-lib/blob/c50bf91dbdd88a83e2f63cef0fb5d115a5ce3b0e/classes/xslt/XMLTypeDescription.inc.php#L135

By
```php
error_log(print_r($errors = libxml_get_errors(), true));
```

Then try to export the problematic issue using the Tools > Import/Export > Native XML Plugin > Export Issues.

Send us the error that you see on the screen + any helpful error message that you may find at the server's error log.

**Comment by janevsky** [comment]
@jonasraoni I did what you suggested, here are the logs:

php error log:
`[26-Jan-2024 10:27:19 Europe/Ljubljana] PHP Fatal error:  Uncaught TypeError: DOMNode::appendChild(): Argument #1 ($node) must be of type DOMNode, null given in /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php:54
Stack trace:
#0 /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php(54): DOMNode->appendChild()
#1 /home/XXX/domains/XXX/public_html/lib/pkp/classes/filter/Filter.inc.php(449): SubmissionFileNativeXmlFilter->process()
#2 /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionNativeXmlFilter.inc.php(149): Filter->execute()
#3 /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionNativeXmlFilter.inc.php(101): SubmissionNativeXmlFilter->addFiles()
#4 /home/XXX/domains/XXX/public_html/plugins/importexport/native/filter/ArticleNativeXmlFilter.inc.php(49): SubmissionNativeXmlFilter->createSubmissionNode()
#5 /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionNativeXmlFilter.inc.php(65): ArticleNativeXmlFilter->createSubmissionNode()
#6 /home/XXX/domains/XXX/public_html/lib/pkp/classes/filter/Filter.inc.php(449): SubmissionNativeXmlFilter->process()
#7 /home/XXX/domains/XXX/public_html/plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php(203): Filter->execute()
#8 /home/XXX/domains/XXX/public_html/plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php(108): IssueNativeXmlFilter->addArticles()
#9 /home/XXX/domains/XXX/public_html/plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php(62): IssueNativeXmlFilter->createIssueNode()
#10 /home/XXX/domains/XXX/public_html/lib/pkp/classes/filter/Filter.inc.php(449): IssueNativeXmlFilter->process()
#11 /home/XXX/domains/XXX/public_html/plugins/importexport/native/NativeImportExportPlugin.inc.php(289): Filter->execute()
#12 /home/XXX/domains/XXX/public_html/plugins/importexport/native/NativeImportExportPlugin.inc.php(212): NativeImportExportPlugin->exportIssues()
#13 /home/XXX/domains/XXX/public_html/lib/pkp/pages/management/PKPToolsHandler.inc.php(94): NativeImportExportPlugin->display()
#14 /home/XXX/domains/XXX/public_html/lib/pkp/classes/core/PKPRouter.inc.php(395): PKPToolsHandler->importexport()
#15 /home/XXX/domains/XXX/public_html/lib/pkp/classes/core/PKPPageRouter.inc.php(246): PKPRouter->_authorizeInitializeAndCallRequest()
#16 /home/XXX/domains/XXX/public_html/lib/pkp/classes/core/Dispatcher.inc.php(144): PKPPageRouter->route()
#17 /home/XXX/domains/XXX/public_html/lib/pkp/classes/core/PKPApplication.inc.php(363): Dispatcher->dispatch()
#18 /home/XXX/domains/XXX/public_html/index.php(68): PKPApplication->execute()
#19 {main}
  thrown in /home/XXX/domains/XXX/public_html/lib/pkp/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php on line 54`

Apache logs:
`[26/Jan/2024:10:27:14 +0100] "GET /XXX/management/importexport/plugin/NativeImportExportPlugin HTTP/1.1" 200 6504 "https://XXX.XXX/XXX/management/tools" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
[26/Jan/2024:10:27:14 +0100] "GET /XXX/$$$call$$$/grid/issues/exportable-issues-list-grid/fetch-grid?_=1706261234599 HTTP/1.1" 200 1544 "https://XXX.XXX/XXX/management/importexport/plugin/NativeImportExportPlugin" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
[26/Jan/2024:10:27:14 +0100] "GET /XXX/api/v1/submissions?searchPhrase=&count=100&offset=0&_=1706261234600 HTTP/1.1" 200 14578 "https://XXX.XXX/XXX/management/importexport/plugin/NativeImportExportPlugin" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
[26/Jan/2024:10:27:18 +0100] "POST /XXX/management/importexport/plugin/NativeImportExportPlugin/exportIssues HTTP/1.1" 500 0 "https://XXX.XXX/XXX/management/importexport/plugin/NativeImportExportPlugin" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
[26/Jan/2024:10:27:47 +0100] "GET /XXX/management/importexport/plugin/NativeImportExportPlugin/exportIssues HTTP/1.1" 500 260 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"`

it was a 500 error, I reloaded and got this
![image](https://github.com/pkp/pkp-lib/assets/63719545/dec207f7-6946-4165-a2c3-da73ba9c62f3)

**Comment by jonasraoni** [comment]
Thanks @janevsky! I see your problem will be addressed by the issue https://github.com/pkp/pkp-lib/issues/8627, which was included in OJS 3.3.0-15.

I recommend you upgrade to the latest OJS 3.3 (3.3.0-16), as it includes also security fixes.
